packages/context/src/binding.ts

```diff
@@ -95,6 +96,24 @@ export class Binding {
   }
 
   /**
+   * Bind the key to a MetaValueProvider
+   * @param providerClass {Function} The MetaValueProvider class (constructor function)
+   * @example
+   * ```ts
+   * ctx.bind('now').toProvider(MyProvider);
+   *
+   * ```
+   */
+  public toProvider<T>(providerClass: Constructor<MetaValueProvider>): this {
+    this.valueConstructor = providerClass;
+    this.getValue = (ctx): Object | Function => {
```

Comment:
The method `getValue` has the following contract (see L43/44):

```ts
getValue(ctx: Context): BoundValue | Promise<BoundValue>
```

Which means your arrow function should be declared as returning `BoundValue | Promise<BoundValue>` too.